facet-ansi/src/lib.rs

```diff
@@ -1,529 +1,400 @@
+    fn with_green(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::Green)))
+    }
 
-        if self.bold {
-            format_codes.push_str("\x1b[1m");
-        }
+    fn with_blue(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::Blue)))
+    }
 
-        if self.underline {
-            format_codes.push_str("\x1b[4m");
-        }
+    fn with_yellow(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::Yellow)))
+    }
 
-        if self.dimmed {
-            format_codes.push_str("\x1b[2m");
-        }
+    fn with_magenta(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::Magenta)))
+    }
 
-        if let Some(color) = self.color {
-            format_codes.push_str(color.ansi_code());
-        }
+    fn with_cyan(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::Cyan)))
+    }
+
+    fn with_white(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::White)))
+    }
 
-        format_codes
+    fn with_black(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::Black)))
+    }
+
+    fn with_bright_red(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::BrightRed)))
+    }
+
+    fn with_bright_green(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::BrightGreen)))
+    }
+
+    fn with_bright_blue(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::BrightBlue)))
+    }
+
+    fn with_bright_yellow(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::BrightYellow)))
+    }
+
+    fn with_bright_magenta(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(
+            anstyle::AnsiColor::BrightMagenta,
+        )))
+    }
+
+    fn with_bright_cyan(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::BrightCyan)))
+    }
+
+    fn with_bright_white(self) -> Style {
+        self.fg_color(Some(anstyle::Color::Ansi(anstyle::AnsiColor::BrightWhite)))
     }
 }
 
-impl<T: Display> Display for StyledDisplay<T> {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        let format_codes = self.style.format_codes();
+/// A struct that wraps a value and its style.
+pub struct Styled<T> {
+    value: T,
+    style: Style,
+}
 
-        if format_codes.is_empty() {
+impl<T: Display> Display for Styled<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        if self.style == Style::new() {
             write!(f, "{}", self.value)
         } else {
-            write!(f, "{}{}\x1b[0m", format_codes, self.value)
+            // anstyle's Style implements Display which handles all the formatting
+            write!(f, "{}{}{}", self.style, self.value, anstyle::Reset)
         }
     }
```

Comment:
I believe this would work:
```rust
fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
    write!(f, "{}{}{:#}", self.style, self.value, self.style)
}
```